app/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -1326,6 +1326,158 @@ def test_serializer_feature_config_validation_bad_json_value(self):
+                NimbusExperiment.Version.FIREFOX_101,
+                False,
+                True,
+            ),
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_105,
+                True,
+                True,
+            ),
+        ]
+    )
+    def test_serializer_feature_enable_version_support_reference_branch(
+        self, application, firefox_version, feature_enabled, valid
+    ):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            status=NimbusExperiment.Status.DRAFT,
+            firefox_min_version=firefox_version,
+            application=application,
+            channel=NimbusExperiment.Channel.NO_CHANNEL,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
+            feature_configs=[
+                NimbusFeatureConfigFactory.create(
+                    schema=None,
+                    application=application,
+                )
+            ],
+        )
+        for feature_value in experiment.reference_branch.feature_values.all():
+            feature_value.enabled = feature_enabled
+            feature_value.value = (
+                """\
+            {
+            "bar": {
+                "baz": "baz",
+                "qux": 123
+            }
+            }
+        """.strip()
+                if feature_enabled
+                else ""
+            )
+            feature_value.save()
+
+        serializer = NimbusReviewSerializer(
+            experiment,
+            data=NimbusReviewSerializer(
+                experiment,
+                context={"user": self.user},
+            ).data,
+            context={"user": self.user},
+        )
+        self.assertEqual(serializer.is_valid(), valid)
+        if not valid:
+            self.assertEqual(len(serializer.errors), 1)
+            error = serializer.errors["reference_branch"].get("feature_enabled")
+            self.assertEqual(error, NimbusConstants.ERROR_FEATURE_ENABLED)
+
+    @parameterized.expand(
+        [
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_105,
+                False,
+                False,
+            ),
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_101,
+                True,
+                True,
+            ),
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_101,
+                False,
+                True,
+            ),
+            (
+                NimbusExperiment.Application.DESKTOP,
+                NimbusExperiment.Version.FIREFOX_105,
+                True,
+                True,
+            ),
+        ]
+    )
+    def test_serializer_feature_enable_version_support_treatment_branches(
```

Comment:
I feel like you could fold these two tests together since they use the same setup.  I know I'm always rambling about not testing too much in a single test, but I feel like these are testing essentially the same thing.  Not a hard requirement, just a suggestion.

Reply:
I completely agree with your point, initially I kept one but then I thought to separate out signifying two different functions in the serializer validating that depending on the type of branch